### PR TITLE
staking apy update

### PR DESCRIPTION
**Pull request by 0xModene**
staking apy math is a bit more accurate now, basing the total off of what's being staked, not whats in Uniswap

### CODE REVIEW

File: src/contexts/Prices/PricesProvider.tsx [hunk 1]
```diff
@@ -67,32 +83,74 @@ const PricesProvider: React.FC = ({ children }) => {
...
 
-    setMviRewardsApy(calculatedApy.toFixed(2))
-  }, [usdInEthMviPool, indexPrice])
+    //get usd price per lp token
+    const pricePerLPToken = new BigNumber(usdInEthMviPool).dividedBy(
+      new BigNumber(totalSupplyInEthMviPool)
+    )
+
+    //multiply by totalSupply
+    getAmountOfStakedTokens(ethereum, mviStakingRewardsAddress).then(
```
Comment (controtie) [inline src/contexts/Prices/PricesProvider.tsx:140]: Are there any problems if `getAmountOfStakedTokens` request fails? Do we need to catch anything?

**Review by controtie — APPROVED** [review]
Looks mostly good. Something to think on: how can we improve separation of concerns in these providers. Not sure if that's against the ethos of react hooks but some of these files can get busy.

File: src/contexts/Prices/PricesProvider.tsx (hunk 1 above)
Comment (0xModene) [reply src/contexts/Prices/PricesProvider.tsx:140]: Great catch, I'll provide some error handling today. There's not a lot we can do data-wise if that happens, but we can zero out everything and log an error.

File: src/contexts/Prices/PricesProvider.tsx [hunk 2]
```diff
@@ -113,8 +113,11 @@ const PricesProvider: React.FC = ({ children }) => {
         const calculatedApy = dailyYield.multipliedBy(365)
 
         setFarmTwoApy(calculatedApy.toFixed(2))
-      }
-    )
+      })
+      .catch((error) => {
+        console.log(error)
+        setFarmTwoApy('0.00')
```
Comment (0xModene) [inline src/contexts/Prices/PricesProvider.tsx:119]: @controtie How's this? Not much else we can do here except fail gracefully

Comment (controtie) [reply src/contexts/Prices/PricesProvider.tsx:119]: Yeah I think that should be good for now

**Pull request merged**